python/lsst/ts/externalscripts/comcam/make_comcam_bias.py

```diff
@@ -0,0 +1,157 @@
+        )
+        self.comcam = ComCam(domain=self.domain, log=self.log)
+        self.ocps = salobj.Remote(domain=self.domain, name="OCPS")
+
+    @classmethod
+    def get_schema(cls):
+        schema = """
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: https://github.com/lsst-ts/ts_externalscripts/blob/master/python/lsst/ts/externalscripts/>-
+        comcam/make_comcam_bias.py
+        title: MakeComCamBias v1
+        description: Configuration for making a LSSTComCam master bias SAL Script.
+        type: object
+        additionalProperties: false
+        required: [instrument, inputCollections]
+        properties:
+            nBias:
+                type: integer
+                default: 1
+                description: number of biases to take
+
+            detectors:
+                type: tuple
+                items:
+                    type: integer
+                    minItems: 1
+                default: (0)
+                descriptor: Detector IDs
+
+            instrument:
+                type: string
+                enum: ["LSSTcComCam", "LATISS", "LSSTCam"]
+                descriptor: Camara name. Must be LSSTComca for now (future: also LATISS or LSSTCam).
+
+            inputCollections:
+                type: string
+                descriptor: Input collectiosn to pass to the bias pipetask.
+        """
+        return yaml.safe_load(schema)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        # Log information about the configuration
+
+        self.log.debug(
+            f"nBias: {config.nBias}, detectors: {config.detectors}, "
+            f"instrument: {config.instrument}. "
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        # Temporary number
+        metadata.duration = 10
+
+    async def run(self):
+        # Should ComCam and OCPS not be enabled here in the script?
+        # original notebook says: "When writing a script, the components
+        # should (probably) be enabled by a user."
+        await self.comcam.enable()
+        await salobj.set_summary_state(self.ocps, salobj.State.ENABLED,
+                                       settingsToApply="LSSTComCam.yaml")
+
+        # Take config.nBiases biases, and return a list of IDs
+        tempBiasList = await self.comcam.take_bias(self.config.nBias)
+        exposures = tuple(tempBiasList)
+        # Bias IDs
+        await self.checkpoint(f"Biases taken: {tempBiasList}")
+
+        # did the images get archived and are they available to the butler?
+        val = await self.comcam.rem.ccarchiver.evt_imageInOODS.aget(timeout=10)
```

Comment:
If I understood correctly you want to wait for all the images to get ingested. I imagine that if you make sure the "last" image was ingested you can assume "all" images where ingested.  In this case, you need to verify that the image you got here is really the last one.. and if it is not, you would need to wait for another event until you got the one you need.

Reply:
I'll check if the retrieved ingested ID corresponds to the last image ID; if not, I'll wait for a couple more events more. If it still doesn't coincide, I'll log a warining.